### PR TITLE
hv: vmsr: add IA32_MISC_ENABLE to msr store area

### PR DESCRIPTION
**Pull request by binbinwu1**
Currently MSR IA32_MISC_ENABLE is passthrough to guest.
However, guest may change the value of this MSR, which will cause issue in hypervisor.
This patch uses VMX MSR store area to isolate the MSR IA32_MISC_ENABLE between guest and host.

TODO:
Some bits of the MSR IA32_MISC_ENABLE is not just per core, but per package.
So need to check if need to prevent guest from setting or clearing these bits that may affect other cores.

Tracked-On: #2834
Signed-off-by: Binbin Wu <binbin.wu@intel.com>
Acked-by: Anthony Xu <anthony.xu@intel.com>

### CODE REVIEW

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by wenlingz** [comment]
@dongyaozu please help approve it, thanks

File: hypervisor/arch/x86/guest/vmsr.c [hunk 1]
```diff
@@ -332,6 +332,11 @@ static void init_msr_area(struct acrn_vcpu *vcpu)
 	vcpu->arch.msr_area.guest[MSR_AREA_TSC_AUX].value = vcpu->vcpu_id;
 	vcpu->arch.msr_area.host[MSR_AREA_TSC_AUX].msr_num = MSR_IA32_TSC_AUX;
 	vcpu->arch.msr_area.host[MSR_AREA_TSC_AUX].value = vcpu->pcpu_id;
+
+	vcpu->arch.msr_area.guest[MSR_AREA_IA32_MISC_ENABLE].msr_num = MSR_IA32_MISC_ENABLE;
```
Comment (dongyaozu) [inline hypervisor/arch/x86/guest/vmsr.c]: can we rename msr_num to msr_index? the later one is more accurate.

Comment (binbinwu1) [reply hypervisor/arch/x86/guest/vmsr.c]: ok

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Pull request merged**